onnx/defs/math/defs.cc

```diff
@@ -3106,12 +3106,16 @@ ONNX_OPERATOR_SET_SCHEMA(
             if (is_onesided || ctx.hasInput(dft_length_arg_index)) {
               // We don't know which axis is the signal dimension, so we cannot infer shape
               // when onesided is enabled or when dft_length_arg_index is provided
-              result_shape_proto.clear_dim();
-
+              TensorShapeProto new_shape_proto{};
+              for (int i = 0; i < rank; ++i) {
+                new_shape_proto.add_dim();
+              }
               // Coerce the last dimension to 2.
-              int dim_size = result_shape_proto.dim_size();
-              result_shape_proto.mutable_dim(dim_size - 1)->set_dim_value(2);
-              updateOutputShape(ctx, output_index, result_shape_proto);
+              ONNX_ASSERTM(
```

Comment:
Was this meant for debugging? Seems a very local, straightforward, assertion, wonder if we need it. (nit)

Reply:
It was meant to guard a segfault because `new_shape_proto.mutable_dim()` will not fail gracefully